src/Process/Factory/InitialTestsRunProcessFactory.php

```diff
@@ -77,7 +79,7 @@ public function createProcess(
 
         $process->setTimeout(null); // Ignore the default timeout of 60 seconds
 
-        if (method_exists($process, 'inheritEnvironmentVariables')) {
+        if (method_exists($process, 'inheritEnvironmentVariables') && version_compare(Versions::getVersion('symfony/console'), 'v4.4', '<')) {
```

Comment:
why do we need to have an additional condition which uses ocramius/package-versions here? does `method_exists` not cover our case?

Reply:
Tried to explain it in the commit message:

> Call `$process->inheritEnvironmentVariables();` only when it exists AND is not deprecated
> Otherwise, it does not make sense because starting from `syfmony/console` 4.4 environment variable are inhereted automatically, and calling deprecated methods fails the tests

We have a job that does `composer install ... --prefer-lowest`. After some dependencies upgrade, this job started installing `symfony/console:4.4.0` instead of `symfony/console:3.*`.

In `symfony/console:4.4.*` the method  `inheritEnvironmentVariables()` exists, but calling it produces a deprecation notice, which [failed the build](https://github.com/infection/infection/runs/3257362627#step:7:134)

So, technically we need to call this method when it exists and and it does something, but in `4.4.*` id does nothing (see the deprecation message: `...env variables are always inherited`).

The final result

* < 4.4.0 - `inheritEnvironmentVariables()` method exists and we should call it
* = `4.4.*` - `inheritEnvironmentVariables()` method exists but we should not call it
* > `4.4.*` - `inheritEnvironmentVariables()`method does not exist

Reply:
ah. I've forgot that we still support symfony packages < 4.4.